### PR TITLE
LPS-175821 create a new paramConverter to check if the scopeKey is valid

### PR DESCRIPTION
**Pull request by jgarcialr**
Hi Team.

This PR aims to solve [LPS-175821](https://issues.liferay.com/browse/LPS-175821).
The problem is  doing a request for a object entry with site scope, via  /scopes/{scopeKey} method  doesn't return a 404 is the scopeKey param is invalid.

I don't know if adding a new ParamConverter is too much or if this should be handled in a different layer, please tell me what do you think and if I have to do any change for this solution.

Thanks in advance.

### CODE REVIEW

**Comment by jgarcialr** [comment]
ci:test:sf

**Comment by jgarcialr** [comment]
ci:test:relevant

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/param/converter/provider/ScopeKeyParamConverterProvider.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import javax.ws.rs.core.UriInfo;
+import javax.ws.rs.ext.ParamConverter;
+import javax.ws.rs.ext.ParamConverterProvider;
+import javax.ws.rs.ext.Provider;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.cxf.jaxrs.utils.AnnotationUtils;
+
+/**
+ * @author Javier Gamarra
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/param/converter/provider/ScopeKeyParamConverterProvider.java:38]: Ups! Wrong author

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/feature/VulcanFeature.java [hunk 2]
```diff
@@ -174,6 +175,8 @@ _roleLocalService, _getScopeChecker(), _sortParserProvider,
 		featureContext.register(
 			new SiteParamConverterProvider(
 				_depotEntryLocalService, _groupLocalService));
+		featureContext.register(
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/feature/VulcanFeature.java:178]: By including this ConverterProvider in VulcanFeature, validations will be performed for all the APIs created by REST Builder, not only Object's one.

Vulcan does not know about Objects

Comment (jgarcialr) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/feature/VulcanFeature.java:178]: Do we have any other use of scopeKey in rest layer?
Looking into the code, I just see this as object related.

**Comment by ccorreagg** [comment]
We have considered to make some changes, as extra validations and also move to the Object's modules. Followed-up PR: https://github.com/liferay-headless/liferay-portal/pull/1034

**Pull request closed without merging**